### ISSUE TITLE
Appstore submission issue - Invalid privacy manifest

### ISSUE THREAD

**Issue by lahirupintospt** [rwbutler/Connectivity#90]
We have received an Appstore email mentioning "Invalid privacy manifest" related to the recent build upload with Connectivity version 8.0. Hopefully its due to the empty array in the PrivacyInfo

Email Info - 
Although submission for App Store review was successful, you may want to correct the following issues in your next submission for App Store review. Once you've corrected the issues, upload a new binary to App Store Connect.

ITMS-91108: Invalid privacy manifest - The PrivacyInfo.xcprivacy file from the following path is invalid: “Frameworks/Connectivity.framework/Connectivity_Privacy.bundle/PrivacyInfo.xcprivacy”. In addition to the privacy manifest files in the locations outlined in the documentation, starting November 12, 2024, all privacy manifests you submit must have valid content. Keys and values in any privacy manifest must be in a valid format. For more details about privacy manifest files, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files and https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/adding_a_privacy_manifest_to_your_app_or_third-party_sdk.

**Comment by GauravWhiz** [comment]
We received this issue and Apple has rejected the app. We got below response:

ITMS-91056: Invalid privacy manifest - The PrivacyInfo.xcprivacy file from the following path is invalid: “Frameworks/Connectivity.framework/Connectivity_Privacy.bundle/PrivacyInfo.xcprivacy”. Keys and values in your app’s privacy manifests must be valid. For more details about privacy manifest files, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files.

Please fix the issue.

**Comment by lahirupintospt** [comment]
Now we are unable to submit new releases to Appstore due to this issue
"Please correct the following issues and upload a new binary to App Store Connect."

Please provide a fix for this :)

**Comment by AkshayUdupa** [comment]
@rwbutler, I'm facing the same issue. Do you have any update on this?

**Comment by rwbutler** [comment]
Thanks for tagging me I'll take a look this evening (also f you're happy to raise a PR that's usually fastest)

**Comment by itsthisjustin** [comment]
We are stuck unable to update our app because of this :(

**Comment by rwbutler** [comment]
@itsthisjustin I don't believe I can add a `NSPrivacyAccessedAPIType` key for any of the APIs covered in [https://developer.apple.com/documentation/bundleresources/describing-use-of-required-reason-api](https://developer.apple.com/documentation/bundleresources/describing-use-of-required-reason-api) since the framework doesn't use any of these. It seems that the options available are:

- Remove the privacy manifest since the framework doesn't actually use file timestamp APIs, system boot time APIs, desk space APIs, active keyboard APIs or User Defaults.
- Remove Reachability

I'll remove the privacy manifest since I can't see a reason why it should be required currently.

I'll also seek to remove Reachability framework likely this weekend.

**Comment by rwbutler** [comment]
Have pushed 8.0.1 now, hope this helps. Please could you let me know how you get on?

**Comment by itsthisjustin** [comment]
In case it helps, we ended up temporarily switching to this fork which worked. https://github.com/Cosmos-Entity/Connectivity

**Comment by rwbutler** [comment]
Thanks @itsthisjustin it's certainly interesting to hear - your app has been accepted using that manifest? I've noted that it's still essentially an empty manifest without any reason strings present so I think the solution for now is to remove it entirely. I've spoken to engineers who are using the version prior to the manifest being added and they haven't experienced any issues submitting to the store.

**Comment by AkshayUdupa** [comment]
I downgraded the version to 6.1.1 and it worked. The is submitted and released. Now will check with the updated version.

**Comment by rwbutler** [comment]
Thanks for the feedback @AkshayUdupa, looking forward to hearing how it goes

**Comment by rwbutler** [comment]
Not had any more feedback so closing this issue as resolved for now.

**Comment by moslienko** [comment]
We have the same problem in 8.0.0.

> Invalid privacy manifest - The PrivacyInfo.xcprivacy file from the following path is invalid: “Frameworks/Connectivity.framework/Connectivity_Privacy.bundle/PrivacyInfo.xcprivacy”. Keys and values in your app’s privacy manifests must be valid.
